controllers/external/util.go

```diff
@@ -169,6 +179,20 @@ func GenerateTemplate(in *GenerateTemplateInput) (*unstructured.Unstructured, er
 	return to, nil
 }
 
+// SetCloneRefAnnotation sets TemplateClonedFromNameAnnotation, TemplateClonedFromGroupKindAnnotation annotations for the cloned infrastructure machine.
+func SetCloneRefAnnotation(ref corev1.ObjectReference, obj *unstructured.Unstructured) error {
```

Comment:
I don't believe this function ever returns a non-nil error. Can we switch to a no-value return?

Reply:
Also, should we inline this directly in `GenerateTemplate` like we do for labels, instead of creating a separate function?